app/src/androidTest/java/com/github/onlynotesswent/ui/EndToEndTest.kt

```diff
@@ -227,17 +293,193 @@ class EndToEndTest {
+  // profile
+  private fun testEndToEndFlow2_init() {
+    `when`(userRepository.getAllUsers(any(), any())).thenAnswer { invocation ->
+      val onSuccess = invocation.getArgument<(List<User>) -> Unit>(0)
+      onSuccess(testUsers)
+    }
+
+    // Mock the user repository to return the specified user
+    `when`(userRepository.getUserById(any(), any(), any(), any())).thenAnswer {
+      val onSuccess = it.arguments[1] as (User) -> Unit
+      val onNotFound = it.arguments[2] as () -> Unit
+      val uid = it.arguments[0] as String
+
+      uidToUser(uid)?.let { it1 -> onSuccess(it1) } ?: onNotFound()
+    }
+
+    `when`(userRepository.getUsersById(any(), any(), any())).thenAnswer {
+      val onSuccess = it.arguments[1] as (List<User>) -> Unit
+      val userIds = it.arguments[0] as List<String>
+
+      onSuccess(userIds.mapNotNull(uidToUser))
+    }
+
+    // Mock add user to initialize current user
+    `when`(userRepository.addUser(any(), any(), any())).thenAnswer {
+      val onSuccess = it.arguments[1] as () -> Unit
+      onSuccess()
+    }
+    // Initialize current user
+    userViewModel.addUser(testUser1, {}, {})
+
+    `when`(userRepository.addFollowerTo(any(), any(), any(), any())).thenAnswer {
+      val onSuccess = it.arguments[2] as () -> Unit
+      val userId = it.arguments[0] as String // testUser2
+      val followerId = it.arguments[1] as String // testUser
+      testUser2 =
+          testUser2.copy(
+              friends =
+                  Friends(
+                      testUser2.friends.following, testUser2.friends.followers.plus(followerId)))
+      testUser1 =
+          testUser1.copy(
+              friends =
+                  Friends(testUser1.friends.following.plus(userId), testUser1.friends.followers))
+
+      onSuccess()
+    }
+
+    `when`(userRepository.removeFollowerFrom(any(), any(), any(), any())).thenAnswer {
+      val onSuccess = it.arguments[2] as () -> Unit
+      val userId = it.arguments[0] as String // testUser2
+      val followerId = it.arguments[1] as String // testUser
+      testUser2 =
+          testUser2.copy(
+              friends =
+                  Friends(
+                      testUser2.friends.following, testUser2.friends.followers.minus(followerId)))
+      testUser1 =
+          testUser1.copy(
+              friends =
+                  Friends(testUser1.friends.following.minus(userId), testUser1.friends.followers))
+      onSuccess()
+    }
+
+    `when`(userRepository.updateUser(any(), any(), any())).thenAnswer {
+      val onSuccess = it.arguments[1] as () -> Unit
+      onSuccess()
+    }
+
+    // Start at overview screen
+    composeTestRule.runOnUiThread { navController.navigate(Route.OVERVIEW) }
+  }
+
+  // Test the end-to-end flow of searching for testUser2 and viewing their profile and following
+  // them and then going to profile screen to unfollow and modify profile
+  @Test
+  fun testEndToEndFlow2() {
```

Comment:
nice looking test. nothing to say. well written, clear and goes through the new features